woocommerce/woocommerce-android
# Enable hiding and showing onboarding list on demand

Closes: #8900 

### Description
Enable users to show or hide the onboarding list on demand: 

- Design details: VyLr7LvKodHE4kINfBE7Lw-fi-4028_120388

![Screenshot 2023-04-26 at 20 37 47](https://user-images.githubusercontent.com/2663464/234671353-f2f659e8-0040-4763-82d0-148c629c228b.png)


### Testing instructions

1. Log into a site that has the onboarding list visible
2. Click on the over flow menu and select "Hide store setup list
3. Notice how the store setup is hidden and the following event is tracked listing all the tasks that are pending and the `source` property with `onboarding_list` value: 
`store_onboarding_hide_list, Properties: {"source":"onboarding_list","hide":true,"pending_tasks":"products, add_domain, payments",`  
4. Navigate to Menu -> Settings -> Store Setup List. Check the "switch" is disabled as the onboarding list is currently hidden
5. Change the "switch" state to re-enable the onboarding list and verify the event is tracked: 
`store_onboarding_hide_list, Properties: {"source":"settings","hide":false,"pending_tasks":"products, add_domain, payments"`
6. Navigate back to my store tab and check onboarding list is displayed again. 
7. Complete all the onboarding tasks and check that the setting to show/hide the onboarding list is not displayed anymore

### Images/gif


https://user-images.githubusercontent.com/2663464/235171737-8af62c67-6e80-43d3-ab75-5bc35c46e00a.mp4
